File: backend/model/display.go
package model

import "github.com/google/uuid"

type Display struct {
	Model
	// attribute
	Name        string `json:"name"`
	Key         string `json:"key"`
	Description string `json:"description"`
	// fk
	BoxId    uuid.UUID     `json:"boxId"`
	LayoutId uint          `json:"layoutId"`
	WidgetId uint          `json:"widgetId"`
	Widget   WidgetDisplay `json:"widget"`
	// relation
	DisplayData []DisplayData `gorm:"foreignKey:DisplayId" json:"displayData"`
	Layout      Layout
}

File: web/src/types/Display.ts
import { Layout } from './Layout';
import { DataDisplay } from './Data';
import { Widget } from './Widget';

export interface DisplayType {
  id: number;
  createdAt: Date;
  updatedAt: Date;
  name: string;
  key: string;
  description: string;
  BoxId: string;
  layoutId: number;
  widgetId: number;
  displayData: DataDisplay[];
  widget: Widget;
  layout: Layout;
}
